common/persistence/PersistenceManager.php

```diff
@@ -0,0 +1,131 @@
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+namespace oat\generis\persistence;
+
+use oat\oatbox\service\ConfigurableService;
+/**
+ * A persistencefactory
+ *
+ * @author Lionel Lecaque  <lionel@taotesting.com>
+ * @license GPLv2
+ * @package generis
+ *
+ */
+class PersistenceManager extends ConfigurableService
+{
+    const SERVICE_ID = 'generis/persistences';
+
+    const OPTION_PERSISTENCES = 'persistences';
+
+    /**
+     * @var array
+     */
+    private static $driverMap = array(
+        'dbal' => 'common_persistence_sql_dbal_Driver',
+        'dbal_pdo_mysql'  => 'common_persistence_sql_dbal_Driver',
+        'dbal_pdo_sqlite' => 'common_persistence_sql_dbal_Driver',
+        'dbal_pdo_pgsql'  => 'common_persistence_sql_dbal_Driver',
+        'dbal_pdo_ibm'    => 'common_persistence_sql_dbal_Driver',
+        'phpredis'   => 'common_persistence_PhpRedisDriver',
+        'phpfile'    => 'common_persistence_PhpFileDriver',
+        'SqlKvWrapper' => 'common_persistence_SqlKvDriver',
+        'no_storage' => 'common_persistence_InMemoryKvDriver',
+        'no_storage_adv' => 'common_persistence_InMemoryAdvKvDriver'
+    );
+
+    /**
+     * @var array
+     */
+    private $persistences = array();
+
+    /**
+     * Returns TRUE if the requested persistence exist, otherwise FALSE.
+     *
+     * @param string $persistenceId
+     * @return bool
+     */
+    public function hasPersistence($persistenceId)
+    {
+        $persistenceList = $this->getOption(static::OPTION_PERSISTENCES);
+        return isset($persistenceList[$persistenceId]);
+    }
+
+    /**
+     * Registers a new persistence.
+     *
+     * @param string $persistenceId
+     * @param array  $persistenceConf
+     */
+    public function registerPersistence($persistenceId, array $persistenceConf)
+    {
+        // wrap pdo drivers in dbal
+        if (strpos($persistenceConf['driver'], 'pdo_') === 0) {
+            $persistenceConf = [
+                'driver' => 'dbal',
+                'connection' => $persistenceConf,
+            ];
+        }
+
+        if (isset($persistenceConf['connection']['driver']) && $persistenceConf['connection']['driver'] === 'pdo_mysql') {
+            $persistenceConf['connection']['charset'] = 'utf8';
+        }
+
+        $configs = $this->getOption(self::OPTION_PERSISTENCES);
+        $configs[$persistenceId] = $persistenceConf;
+        $this->setOption(self::OPTION_PERSISTENCES, $configs);
+    }
+
+    /**
+     * @return \common_persistence_Persistence
+     */
+    public function getPersistenceById($persistenceId)
+    {
+        if(!isset($this->persistences[$persistenceId])) {
+            $this->persistences[$persistenceId] = $this->createPersistence($persistenceId);
+        }
+        return $this->persistences[$persistenceId];
+    }
+
+    /**
+     * @param string $persistenceId
+     * @throws \common_Exception
```

Comment:
Please use `use` statements to avoid `\` in code hopefully with alias like:
```
use \common_Exception as CommonException;
use \common_persistence_Persistence as Persistence;
```